### DISCUSSION TITLE
Creating initrd for efi shell with multiple files

### DISCUSSION THREAD

**Discussion by rbraun204** [ipxe/ipxe#668] [General]
Hey everyone,  just wrapping my head around getting an efi shell up and working and having access to the efi applications I need to use flashing bioses on HP desktops.

I've built my ipxe images,   everything is generally working ipxe wise,  but I'm having trouble creating an initrd archive of the around 10 or so files I would like to have access to from the efi shell.

When I launch my shell via

initrd http://${server-ip}/netboot/efi/CustomLogoApp.efi
#initrd http://${server-ip}/netboot/efi/hpefiutils.initrd
chain http://${server-ip}/netboot/efi/efi_shell_x64.efi

I can boot into the shell and see and run CustomLogoApp.efi fine.    When I flip the commented sections around and boot with the hpefiutils.initrd,   I can now see the whole hpefiutils.initrd and it is not unpacked.

Not sure if I'm doing the initrd call incorrect,   or possibly created bad cpio archive with the following command,  or do we just stack initrd calls for each file we want in the shell, or a recursive add?

init creation

cd /tmp/HP
find . -print0 | cpio --null --create --format=newc > /tmp/hpefiutils.initrd

Thanks all!

**Comment by mcb30** [comment]
There is (currently) no capability within either iPXE or the UEFI firmware for unpacking a CPIO archive.  You will need to download the files individually in order to have them available to the UEFI shell.

Hope that helps!

**Comment by rbraun204** [comment]
Roger that thanks for the help!
